src/androidbackup_fmt_plug.c

```diff
@@ -0,0 +1,393 @@
+/*
+ * Format for cracking encrypted Android Backups.
+ *
+ * This software is Copyright (c) 2018, Dhiru Kholia <dhiru [at] openwall.com>,
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_ab;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_ab);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#define OMP_SCALE               4  // tuned on i7-7820HQ
+
+#include "formats.h"
+#include "misc.h"
+#include "common.h"
+#include "params.h"
+#include "options.h"
+#include "sha2.h"
+#include "aes.h"
+// #define PBKDF2_HMAC_SHA1_ALSO_INCLUDE_CTX 1
+#include "pbkdf2_hmac_sha1.h"
+#include "memdbg.h"
+
+#define FORMAT_LABEL            "AndroidBackup"
+#define FORMAT_NAME             ""
+#define FORMAT_TAG              "$ab$"
+#define TAG_LENGTH              (sizeof(FORMAT_TAG) - 1)
+#ifdef SIMD_COEF_64
+#define ALGORITHM_NAME          "AES PBKDF2-SHA1 " SHA1_ALGORITHM_NAME
+#else
+#define ALGORITHM_NAME          "AES PBKDF2-SHA1 32/" ARCH_BITS_STR
+#endif
```

Comment:
The significant part of this is PBKDF2 so should be listed first. AES can be omitted completely but personally I like it there.
```c
#ifdef SIMD_COEF_64
#define ALGORITHM_NAME          "PBKDF2-SHA1 " SHA1_ALGORITHM_NAME " AES"
#else
#define ALGORITHM_NAME          "PBKDF2-SHA1 32/" ARCH_BITS_STR " AES"
#endif
```

Reply:
Fixed now. Thanks!